content/articles/how-to-implement-caching-in-adonisjs-5/index.md

```diff
@@ -0,0 +1,221 @@
+npm i @kaperskyguru/adonis-cache
+```
+
+Next, run the below command to configure and set up the package.
+
+```bash
+node ace invoke @kaperskyguru/adonis-cache
+# or
+node ace configure @kaperskyguru/adonis-cache
+```
+
+Next, open the `.env` file and inform the cache package to use `redis` as our cache driver. 
+
+```js
+CACHE_DRIVER=redis
+```
+
+Continue below to configure our Redis server, so that our cache package can communicate successfully to the Redis server.
+
+The default driver is set to `file`, but you can read the documentation to understand the different drivers and configure them.
+
+Before using Redis as your driver, make sure to have Redis installed on your machine, and it is also configured correctly to work with Adonis 5.
+
+To install the Redis community server on your local machine, [download the binary file](https://redis.io/download) depending on your OS, and follow the instructions.
+
+The cache package uses the default Redis configuration provided by Adonis 5. Therefore, you must configure your Adonis Redis configuration.
+
+Follow the steps below to install and configure the Redis server for Adonis 5.
+
+### Installing Adonis Redis
+To use `redis` as your cache driver, you need to have it installed on your server and installed and configured in your Adonis 5 project.
+
+To do that, run the following command.
+
+```bash
+npm i @adonisjs/redis
+```
+
+Configure `@adonisjs/redis` with this command.
+
+```bash
+node ace configure @adonisjs/redis
+//Or
+node ace invoke @adonisjs/redis
+```
+
+There are more configurations and tweaks for even better Redis usage than the default options we have just accepted. You can read more about Adonis Redis configuration [here](https://docs.adonisjs.com/guides/redis).
+
+### Handling caching in Adonis 5
+Before implementing caching, you should have [cloned the Forum API project](https://github.com/Kaperskyguru/adonisjs-forum-api). You should also have the Redis server, Adonis Redis, and the caching package we will be using in this project installed and configured.
+
+If you have all these checked, we are ready to dive right into the code.
+
+In the world of caching, there are many caching strategies and methods available depending on the structure of your data structure and the different operations performed on the data. You can choose other caching strategies depending on the features and type of the project.
+
+You can read through the [best caching Strategy for Adonis 5](https://masteringbackend.com/posts/caching-strategy-for-restful-api/) which will show you the different caching strategies, and help you choose which one to use depending on the features or the type of project.
+
+We will be using the **cache aside strategy** for all the read operations and a mixture of **write Around** and **write through strategy** for the write operations.
+
+Open your Controller file. 
+
+We already have the `TicketsController` file found in our previous project's `app/Controllers/Http` folder.
+
```

Comment:
In the specified folder, there's no `TicketsController.ts`. The only file that resembles this is the `ForumsController.ts`.

Reply:
Done